pkg/fanal/test/integration/containerd_test.go

```diff
@@ -75,6 +75,158 @@ func startContainerd(t *testing.T, ctx context.Context, hostPath string) testcon
 	return containerdC
 }
 
+// Each of these tests imports an image and tags it with the name found in the
+// `imageName` field. Then, the containerd store is searched by the reference
+// provided in the `searchName` field.
+func TestContainerd_SearchLocalStoreByNameOrDigest(t *testing.T) {
+	type testInstance struct {
+		name       string
+		imageName  string
+		searchName string
+		digest     bool
+		expectErr  bool
+	}
+
+	tests := []testInstance{
+		{
+			name:       "familiarName:tag",
+			imageName:  "hello:world",
+			searchName: "hello:world",
+		},
+		{
+			name:       "compound/name:tag",
+			imageName:  "say/hello:world",
+			searchName: "say/hello:world",
+		},
+		{
+			name:       "docker.io/library/name:tag",
+			imageName:  "docker.io/library/hello:world",
+			searchName: "docker.io/library/hello:world",
+		},
+		{
+			name:       "other-registry.io/library/name:tag",
+			imageName:  "other-registry.io/library/hello:world",
+			searchName: "other-registry.io/library/hello:world",
+		},
+		{
+			name:       "other-registry.io/library/wrong:wrongTag should fail",
+			imageName:  "other-registry.io/library/hello:world",
+			searchName: "other-registry.io/library/hello:badtag",
+			expectErr:  true,
+		},
+		{
+			name:       "other-registry.io/library/wrongName:tag should fail",
+			imageName:  "other-registry.io/library/hello:world",
+			searchName: "other-registry.io/library/badname:world",
+			expectErr:  true,
+		},
+		{
+			digest:     true,
+			name:       "digest should succeed",
+			imageName:  "",
+			searchName: "sha256:63775b87d672bd68c667a244bf7664a5b26f1f8973662b7dd730e2170b1ff2bf",
```

Comment:
Seems like this digest comes from named images in the previous cases.

```
$ sudo nerdctl images --no-trunc
REPOSITORY                                TAG           IMAGE ID                                                                   CREATED          PLATFORM       SIZE     BLOB SIZE
hello                                     world         sha256:63775b87d672bd68c667a244bf7664a5b26f1f8973662b7dd730e2170b1ff2bf    1 second ago     linux/amd64    0.0 B    2.7 MiB
ghcr.io/aquasecurity/trivy-test-images    alpine-310    sha256:f12582b2f2190f350e3904462c1c23aaf366b4f76705e97b199f9bbded1d816a    5 minutes ago    linux/amd64    0.0 B    2.7 MiB
```

If you run this test only, it fails with the following error.

```
1 error occurred:
	* image not found in containerd store: sha256:63775b87d672bd68c667a244bf7664a5b26f1f8973662b7dd730e2170b1ff2bf
```

I think we should use `f12582b2f2190f350e3904462c1c23aaf366b4f76705e97b199f9bbded1d816a` instead so each test can be independent.

```suggestion
			searchName: "sha256:f12582b2f2190f350e3904462c1c23aaf366b4f76705e97b199f9bbded1d816a",
```

We may want to remove the imported image so that we can make sure each test does not depend on each.

```
t.Cleanup(func() {
	for _, img := range imgs {
		_ = client.ImageService().Delete(ctx, img.Name, images.SynchronousDelete())
	}
})
```

Reply:
Thanks for the review; I'll update shortly

Reply:
Another unintended consequence was that `containerd.WithIndexName(tt.imageName)` was changing the digest of the image with each import, because the name of the original image (`ghcr.io/aquasecurity/trivy-test-images:alpine-310`) gets included in the `annotations` section of the manifest pointed to by `index.json`. That of course changes the hash of the manifest to `63775...`. 

I am working on a different solution that simply re-tags the image after import.

Reply:
I have rebased and pushed changes which resolve the issues you raised. Thanks again for the careful review